bin/p2-exec/main.go

```diff
@@ -114,6 +114,12 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
+
+	// setpgid: zeroes for current process ID
+	if err := unix.Setpgid(0, 0); err != nil {
```

Comment:
Relying on:

* "If pid is zero, then the call applies to the current process"/"if pid is 0, the process ID of the calling process shall be used" (appears on both Linux's and BSD's man page for setpgid)
* "Also, if pgid is 0, the process ID of the indicated process shall be used." (appears only on Linux's man page)

That second bullet point is an interesting question - maybe we need to be explicit, then?